### PR TITLE
[odo] deleting app deletes all components

### PR DESCRIPTION
**Pull request by ghost**
https://github.com/openshift/odo/issues/2180

### CODE REVIEW

**Comment by pmacko1** [comment]
@Preeticp @boczkowska could you please take a look at this draft? Thanks!

File: modules/openshift-developer-cli-deleting-an-application.adoc [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+====
+Suppress the confirmation prompt with `-f` flag.
+====
++
+----
+$ odo app delete <application_name>
+    ? Are you sure you want to delete the application: <application_name> from project: <project name>
+----
+
+. The app is deleted.
```
Comment (Preeticp) [inline modules/openshift-developer-cli-deleting-an-application.adoc]: This is not a step. It is also unnecessary to mention it, suggest removing it.

**Review by Preeticp — COMMENTED** [review]
@pmacko1 @amitkrout can you please confirm that a user would need to run both step 1 and 2 listed here to get a list of his apps and components?

File: modules/openshift-developer-cli-deleting-an-application.adoc (hunk 1 above)
Comment (pmacko1) [reply modules/openshift-developer-cli-deleting-an-application.adoc]: Yes, I was not sure about this one.

**Comment by amitkrout** [comment]
> @pmacko1 @amitkrout can you please confirm that a user would need to run both step 1 and 2 listed here to get a list of his apps and components?

Can you please provide me the link where i can view the impact of this pr in the doc ?

File: modules/openshift-developer-cli-deleting-an-application.adoc (hunk 1 above)
Comment (amitkrout) [reply modules/openshift-developer-cli-deleting-an-application.adoc]: Please remove it.

File: modules/openshift-developer-cli-deleting-an-application.adoc [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+.Procedure
+
+. List the applications in the current project:
++
+----
+$ odo app list
+----
+
+. List the components associated with the applications:
```
Comment (amitkrout) [inline modules/openshift-developer-cli-deleting-an-application.adoc]: This step is unnecessary. Moreover you have already mentioned that ```Deleting the application will delete all components associated with the application.```, so you can remove this step.

**Review by amitkrout — CHANGES_REQUESTED** [review]

File: modules/openshift-developer-cli-deleting-an-application.adoc [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+= Deleting an Application
+
+[IMPORTANT]
+====
+Deleting the application will delete all components associated with the application.
+====
+
+.Procedure
+
+. List the applications in the current project:
```
Comment (amitkrout) [inline modules/openshift-developer-cli-deleting-an-application.adoc]: +1, user should know which app is being deleted.

File: modules/openshift-developer-cli-deleting-an-application.adoc (hunk 2 above)
Comment (pmacko1) [reply modules/openshift-developer-cli-deleting-an-application.adoc]: hm, makes sense, but what if the user does not want those components removed by mistake? Just a fail-safe, maybe.

File: modules/openshift-developer-cli-deleting-an-application.adoc (hunk 1 above)
Comment (pmacko1) [reply modules/openshift-developer-cli-deleting-an-application.adoc]: done!

File: modules/openshift-developer-cli-deleting-an-application.adoc [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+. List the components associated with the applications:
++
+----
+$ odo component list
+    APP     NAME                      TYPE       SOURCE        STATE
+    app     nodejs-nodejs-ex-elyf     nodejs     file://./     Pushed
+----
+
+. Delete the application, confirm the deletion with `Y`:
```
[the remark this answers is not in the document]
Comment (pmacko1) [reply modules/openshift-developer-cli-deleting-an-application.adoc]: done!

File: modules/openshift-developer-cli-deleting-an-application.adoc (hunk 2 above)
Comment (pmacko1) [reply modules/openshift-developer-cli-deleting-an-application.adoc]: as per our convo in slack @amitkrout let's leave this here as a fail-safe for users not really reading the warnings (e.g. me). Just re-assure users that the command doesn't do anything besides listing the components

**Comment by amitkrout** [comment]
@Preeticp Doc change looks good to me. @pmacko1 thanks for the pr :)

File: modules/openshift-developer-cli-deleting-an-application.adoc (hunk 2 above)
Comment (pmacko1) [reply modules/openshift-developer-cli-deleting-an-application.adoc]: On the other hand, let's just say the app deletes the components the second time, as lists do usually what lists do -> list :)

File: modules/openshift-developer-cli-deleting-an-application.adoc [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
++
+[NOTE]
+====
+Suppress the confirmation prompt with `-f` flag.
+====
```
[the remark this answers is not in the document]
Comment (pmacko1) [reply modules/openshift-developer-cli-deleting-an-application.adoc]: fixed!

**Comment by Preeticp** [comment]
@pmacko1 please remove the 'confirm the deletion with `y`' part from step 3, you have now added it to step 4.
Screenshot from 2019-12-05 11-54-59

Once you are done with that, please ask review from the OCP team.

**Comment by pmacko1** [comment]
@bergerhoffer Hi Andrea, can I ask you for a review, please?

File: modules/openshift-developer-cli-deleting-an-application.adoc [hunk 6]
```diff
@@ -0,0 +1,40 @@
+// Module included in the following assemblies:
+//
+// * cli_reference/openshift_developer_cli/creating-a-single-component-application-with-odo.adoc
+// * cli_reference/openshift_developer_cli/creating-an-application-with-a-database.adoc
+// * cli_reference/openshift_developer_cli/creating-a-multicomponent-application-with-odo.adoc
+
+[id="deleting-an-application_{context}"]
+= Deleting an Application
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-deleting-an-application.adoc]: Application -> application (sentence case)

File: modules/openshift-developer-cli-deleting-an-application.adoc [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+[IMPORTANT]
+====
+Deleting the application will delete all components associated with an application.
```
Comment (bergerhoffer) [inline modules/openshift-developer-cli-deleting-an-application.adoc]: Minor nitpick, but I would swap the "the" and "an" before the two application instances:

```
Deleting an application will delete all components associated with the application.
```

**Review by bergerhoffer — COMMENTED** [review]
Just a few comments. I've marked this for 4.2 and beyond (4.3), let me know if this is correct.

Once you've made the last updates, please squash the commits. And be sure to remove the `WIP` once this is ready to be merged. Thanks!

**Comment by pmacko1** [comment]
@bergerhoffer Thanks for the review Andrea, I've squashed the commits. I believe the PR is ready to be merged, removing WIP

**Pull request merged**